### PR TITLE
Various Phase 2 Server Work

### PR DESCRIPTION
**Pull request by dmzimmerman**

### CODE REVIEW

**Comment by kiniry** [comment]
This is labeled `phase-2` in the branch name, but has a milestone of `phase-1`, so I'm not merging it, but only reviewing it.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractEndpoint.java [hunk 1]
```diff
@@ -2,22 +2,19 @@
  * Free & Fair Colorado RLA System
  * 
  * @title ColoradoRLA
- *
```
Comment (kiniry) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractEndpoint.java]: That pesky borked Eclipse reformatter!

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractEndpoint.java [hunk 2]
```diff
@@ -193,8 +197,9 @@ protected boolean transitionAndSaveASM(final Response the_response)  {
    * @param the_response the HTTP response.
    */
   public void ok(final Response the_response) {
-    the_response.status(HttpStatus.OK_200);    
+    the_response.status(HttpStatus.OK_200);   
     my_endpoint_result = "";
+
```
Comment (kiniry) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractEndpoint.java]: Where did this new CR come from?

File: server/eclipse-project/src/main/java/us/freeandfair/corla/model/LogEntry.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+@Table(name = "log")
+//this class has many fields that would normally be declared final, but
+//cannot be for compatibility with Hibernate and JPA.
+@SuppressWarnings("PMD.ImmutableField")
+public class LogEntry extends AbstractEntity implements Serializable {
+  /**
+   * The root hash for the hash chain (a 256-bit block of zeros).
+   */
+  public static final String ROOT_HASH = 
+      "0000000000000000000000000000000000000000000000000000000000000000";
```
Comment (kiniry) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/model/LogEntry.java]: Perhaps we should be cheeky and embed an Easter Egg?

**Comment by dmzimmerman** [comment]
It's also marked "work in progress", and was not ready for review. But that's fine.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/model/LogEntry.java (hunk 3 above)
Comment (dmzimmerman) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/model/LogEntry.java]: Fine by me... any suggestions?

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractEndpoint.java (hunk 1 above)
Comment (dmzimmerman) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractEndpoint.java]: Yes, I know. We'll have to fix it at some point.

**Comment by dmzimmerman** [comment]
Closing and replacing with another PR.

**Pull request closed without merging**